esmvaltool/diag_scripts/sea_surface_salinity/compare_salinity.py

```diff
@@ -0,0 +1,261 @@
+        """
+        years = [cell.point.year for cell in cube.coord('time').cells()]
+        # be extra sure that the first point is not in the previous year
+        if 0 not in np.diff(years):
+            return regrid_time(cube, 'yr')
+        return cube
+
+    def _datetime_to_int_days(self, cube):
+        """Return list of int(days) converted from cube datetime cells."""
+        cube = self._align_yearly_axes(cube)
+        time_cells = [cell.point for cell in cube.coord('time').cells()]
+
+        # extract date info
+        real_dates = []
+        for date_obj in time_cells:
+            # real_date resets the actual data point day
+            # to the 1st of the month so that there are no
+            # wrong overlap indices
+            real_date = datetime(date_obj.year, date_obj.month, 1, 0, 0, 0)
+            real_dates.append(real_date)
+
+        # get the number of days starting from the reference unit
+        time_unit = cube.coord('time').units.name
+        time_offset = self._get_time_offset(time_unit)
+        days = [(date_obj - time_offset).days for date_obj in real_dates]
+
+        return days
+
+    def _get_overlap(self, cubes):
+        """
+        Get discrete time overlaps.
+        This method gets the bounds of coord time
+        from the cube and assembles a continuous time
+        axis with smallest unit 1; then it finds the
+        overlaps by doing a 1-dim intersect;
+        takes the floor of first date and
+        ceil of last date.
+        """
+        all_times = []
+        for cube in cubes:
+            span = self._datetime_to_int_days(cube)
+            start, stop = span[0], span[-1]
+            all_times.append([start, stop])
+        bounds = [range(b[0], b[-1] + 1) for b in all_times]
+        time_pts = reduce(np.intersect1d, bounds)
+        if len(time_pts) > 1:
+            time_bounds_list = [time_pts[0], time_pts[-1]]
+            return time_bounds_list
+
+    def _slice_cube(self, cube, t_1, t_2):
+        """
+        Efficient slicer.
+        Simple cube data slicer on indices
+        of common time-data elements.
+        """
+        time_pts = [t for t in cube.coord('time').points]
+        converted_t = self._datetime_to_int_days(cube)
+        idxs = sorted([
+            time_pts.index(ii) for ii, jj in zip(time_pts, converted_t)
+            if t_1 <= jj <= t_2
+        ])
+        return [idxs[0], idxs[-1]]
+
+
+class TextHandler(HandlerBase):
+    def create_artists(self, legend, text, xdescent, ydescent,
+                       width, height, fontsize, trans):
+        tx = Text(width/2., height/2, text, fontsize=fontsize,
+                  ha="center", va="center", fontweight="bold")
+        return [tx]
+
+
+Legend.update_default_handler_map({str: TextHandler()})
+
+
+def main():
+    with esmvaltool.diag_scripts.shared.run_diagnostic() as config:
+        CompareSalinity(config).compute()
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
OK - I am bit confused why you've created a couple very hefty objects instead of just using functions - it's not like the diagnostic is a module that gets used many times

Reply:
Not sure, but I would guess @jvegasbsc had a good reason...